src/DiagnosticVariables.jl

```diff
@@ -47,13 +54,44 @@ Keyword arguments
 - `comments`: More verbose explanation of what the variable is, or comments related to how
               it is defined or computed.
 """
-Base.@kwdef struct DiagnosticVariable{T <: Function}
-    compute!::T
-    short_name::String = ""
-    long_name::String = ""
-    standard_name::String = ""
-    units::String = ""
-    comments::String = ""
+struct DiagnosticVariable{
+    F1 <: Union{Function, Nothing},
+    F2 <: Union{Function, Nothing},
+}
+    compute!::F1
+    compute::F2
+    short_name::String
+    long_name::String
+    standard_name::String
+    units::String
+    comments::String
+end
+
+function DiagnosticVariable(;
+    compute = nothing,
+    compute! = nothing,
+    short_name = "",
+    long_name = "",
+    standard_name = "",
+    units = "",
+    comments = "",
+)
+    if isnothing(compute) && isnothing(compute!)
+        error("One between `compute` and `compute!` has to be provided")
+    end
+    if !isnothing(compute) && !isnothing(compute!)
+        error("`compute` and `compute!` cannot be both provided")
+    end
```

Comment:
This might a good place to use `xor`?

Reply:
I didn't realize julia had `xor` (but not `or` or `and`...)